eql/parser.py

```diff
@@ -442,7 +442,16 @@ def _update_field_info(self, node_info, optional_syntax=False):
                     return NodeInfo(comparison, TypeHint.Boolean, nullable=not self._strict_fields, source=node_info)
 
         else:
-            type_hint = self._var_types[field.base]
+            # Extract out types from variables and perform validation on the nested schema
+            schema_hint = self._var_types[field.base]
```

Comment:
In which case is this logic hit re arraysearch use?

Reply:
```
arraySearch(someArray, $myvar, $myvar.foo == "bar")
                               ^^^^^^
```